documentation/architecture.md

```diff
@@ -0,0 +1,336 @@
+7. Send the pod creation command to `hyperstart`. The `hyperstart` pod is the container process sandbox.
+8. Send the allocateIO command to the proxy, for getting the `hyperstart` I/O sequence numbers described in step 2.
+9. Pass the `cc-proxy` socketpair file descriptor, and the I/O sequence numbers to the listening cc-shim process through the dedicated pipes.
+10. The `cc-shim` instance is put into a stopped state to prevent it from doing I/O before the container is started.
+
+![Docker create](https://github.com/01org/cc-oci-runtime/blob/topic/documentation/documentation/create.png)
+
+At that point, the container sandbox is created in the virtual machine and `cc-shim` is stopped on the host.
+But the container process itself is not yet running as one needs to call `docker start` to actually start it.
+
+##### `start`
+
+On namespace containers `start` launches a traditional Linux container process in its own set of namespaces.
+With Clear Containers, the main task of `cc-oci-runtime` is to create and start a container within the
+pod that got created during the `create` step. In practice, this means `cc-oci-runtime` follows
+those steps:
+
+1. `cc-oci-runtime` connects to `cc-proxy` and sends it the `attach` command to let it know which pod
+we want to use to create and start the new container.
+2. `cc-oci-runtime` sends an hyperstart `NEWCONTAINER` command to create and start a new container in
+a given pod. The command is sent to `cc-proxy` who forwards it to the right hyperstart instance running
+in the appropriate guest.
+3. `cc-oci-runtime` resumes `cc-shim` so that it can now connect to the `cc-proxy` and acts as
+a signal and I/O streams proxy between `containerd-shim` and `cc-proxy`.
+
+##### `exec`
+
+`docker exec` allows one to run an additional command within an already running container.
+With Clear Containers, this translates into sending a `EXECCMD` command to hyperstart so
+that it runs a command into a running container belonging to a certain pod.
+All I/O streams from the executed command will be passed back to Docker through a newly
+created `cc-shim`.
+
+The `exec` code path is partly similar to the `create` one and `cc-oci-runtime` goes through
+the following steps:
+
+1. `cc-oci-runtime` connects to `cc-proxy` and sends it the `attach` command to let it know which pod
+we want to use to run the `exec` command.
+2. `cc-oci-runtime` sends the allocateIO command to the proxy, for getting the `hyperstart` I/O sequence
+numbers for the `exec` command I/O streams.
+3. `cc-oci-runtime` sends an hyperstart `EXECMD` command to start the command in the right container
+The command is sent to `cc-proxy` who forwards it to the right hyperstart instance running
+in the appropriate guest.
+4. Spawn the `cc-shim` process for it to forward the output streams (stderr and stdout) and the `exec`
+command exit code to Docker.
+
+Now the Clear Container is running as an isolated process in its own virtual machine.
```

Comment:
As we are talking about `exec` here, I think this sentence should be talking about how the exec process is running in, maybe?, its own namespace - or, is now running in the same namespace etc. (that is, describe what is different/the same between the original workload and an exec'd one within the same container).

I suspect this sentence may either be a copy/paste hangover, or it should be more detached from the exec section ?